camel/memories/agent_memories.py

```diff
@@ -0,0 +1,150 @@
+        return self._chat_history_block.retrieve(self._window_size)
+
+    def write_records(self, records: List[MemoryRecord]) -> None:
+        self._chat_history_block.write_records(records)
+
+    def get_context_creator(self) -> BaseContextCreator:
+        return self._context_creator
+
+    def clear(self) -> None:
+        return self._chat_history_block.clear()
+
+
+class VectorDBMemory(AgentMemory):
+    r"""An agent memory wrapper of :obj:`VectorDBBlock`. This memory queries
+    messages stored in the vector database. Notice that the most recent
+    messages will not be added to the context.
+
+    Args:
+        context_creator (BaseContextCreator): A model context creator.
+        storage (BaseVectorStorage, optional): A vector storage storage. If
+            `None`, an :obj:`QdrantStorage` will be used.
+            (default: :obj:`None`)
+        retrieve_limit (int, optional): The maximum number of messages
+            to be added into the context.  (default: :obj:`3`)
+    """
+
+    def __init__(
+        self,
+        context_creator: BaseContextCreator,
+        storage: Optional[BaseVectorStorage] = None,
+        retrieve_limit=3,
+    ) -> None:
+        self._context_creator = context_creator
+        self._retrieve_limit = retrieve_limit
+        self._vectordb_block = VectorDBBlock(storage=storage)
+
+        self._current_topic: str = ""
+
+    def retrieve(self) -> List[ContextRecord]:
+        return self._vectordb_block.retrieve(
+            self._current_topic,
+            limit=self._retrieve_limit,
+        )
+
+    def write_records(self, records: List[MemoryRecord]) -> None:
+        # Assume the last user input is the current topic.
+        for record in records:
+            if record.role_at_backend == OpenAIBackendRole.USER:
+                self._current_topic = record.message.content
+        self._vectordb_block.write_records(records)
+
+    def get_context_creator(self) -> BaseContextCreator:
+        return self._context_creator
+
+
+class LongtermAgentMemory(AgentMemory):
+    r"""An implementation of the :obj:`AgentMemory` abstract base class for
+    augumenting ChatHistoryMemory with VectorDBMemory.
+    """
+
+    def __init__(
+        self,
+        context_creator: BaseContextCreator,
+        chat_history_block: Optional[ChatHistoryBlock] = None,
+        vector_db_block: Optional[VectorDBBlock] = None,
+    ) -> None:
+        self.chat_history_block = chat_history_block or ChatHistoryBlock()
+        self.vector_db_block = vector_db_block or VectorDBBlock()
+        self._context_creator = context_creator
+        self._current_topic: str = ""
+
+    def get_context_creator(self) -> BaseContextCreator:
+        return self._context_creator
+
+    def retrieve(self) -> List[ContextRecord]:
+        chat_history = self.chat_history_block.retrieve()
+        vector_db_retrieve = self.vector_db_block.retrieve(
+            self._current_topic, limit=5)
```

Comment:
This limit is 5 that the setting is a little bit different from other memory, which can specify window_size and retrieve_limit? Maybe keep them consistent is a little bit better?

Reply:
Yes, it makes sense. Thank you!